### ISSUE TITLE
FP S5689 because one "escape check" case is missing

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#2306]
When this rule was implemented we did some "escape checks" because SonarJS is not cross file analysis, [see last comment of the original ticket](https://github.com/SonarSource/SonarJS/issues/1836):
```
I understand the impact of "evidence of absence"-type rules on FPs. I've added an additional "escape check": if app ends up as parameter to some kind of function, as in configureMiddleware(app), then we deactivate this check entirely. This should (hopefully) reduce the FP rate somewhat. At least it helped with one case in integration tests.
```

this kind of "escape check" allows us to check if the express object is likely used in another file and so we don't raise in this case:
```
const appsensitive = express() // Compliant (S5689) because if we find the express object as argument of other function we don't raise

// this function is defined in another file
correctlyConfigS5689(appsensitive);
```
```
module.exports.correctlyConfigS5689 = function(app) {
  
  app.use(hidePoweredBy()) // Compliant solution
};
```

This ticket is about another "escape check" we should implement:
- if the express app is created in a function and the express app object is returned from this function then we should not raise
```
module.exports.createExpressApp = function() {
  
  var appsensitive = express() // should be compliant (but FP for the moment) because express object is returned from the function
  
  return appsensitive;
};
```
```
const appsensitive3 = createExpressApp();
appsensitive3.use(hidePoweredBy()) // Compliant solution

```

Code example: https://github.com/SonarSource/security-expected-issues/blob/master/javascript/express.js/express.config.crossfile.js
